### ISSUE TITLE
'az deployment group validate' returns invalid JSON on validation failure

### ISSUE THREAD

**Issue by peterinnesmsft** [Azure/azure-cli#13037]
## Describe the bug

When attempting to validate an ARM template for deployment, ARM returns a validation failure which contains well-formed JSON. However, when the CLI returns the corresponding error, it is serialized in such a which renders the returned JSON invalid, and strips out the key inner error information which conveys what the particular validation failure was.

**Command Name**
`az deployment group validate`

**Errors:**
```
{'additional_properties': {}, 'code': 'InvalidTemplateDeployment', 'message': "The template deployment 'template' is not valid according to the validation procedure. The tracking id is '62bf6fbb-dc46-4e50-a156-f37966baa63b'. See inner errors for details.", 'target': None, 'details': [<azure.mgmt.resource.resources.v2019_07_01.models._models.ErrorResponse object at 0x056B4370>], 'additional_info': None}
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- Provide input which is bound to trigger a validation failure (e.g. invalid resource name).
- `az deployment group validate --resource-group wer_ter --template-file "C:\path\to\template.json" --parameters <key_value_pairs> --output json --debug`

## Expected Behavior

Per the debug output when invoking the command, the following JSON output is expected to be returned:

`{
	"error": {
		"code": "InvalidTemplateDeployment",
		"message": "The template deployment 'template' is not valid according to the validation procedure. The tracking id is '62bf6fbb-dc46-4e50-a156-f37966baa63b'. See inner errors for details.",
		"details": [
			{
				"code": "CustomDomainInvalid",
				"message": "Sub domain name 'wer_ter-luisruntime-if6lsur' is not valid."
			}
		]
	}
}`

## Environment Summary
```
Windows-10-10.0.18362-SP0
Python 3.6.6
Installer: MSI

azure-cli 2.3.1

Extensions:
azure-devops 0.18.0

```

**Comment by yungezz** [comment]
hi @zhoxing-ms could you pls have a look?

**Comment by yonzhan** [comment]
add to S169

**Comment by zhoxing-ms** [comment]
@peterinnesmsft Hi, this problem has been fixed in this PR #12848, please try it with the latest version of CLI (2.4.0).
If you still have some question, please feel free to let us know~

**Comment by peterinnesmsft** [comment]
@zhoxing-ms thanks for addressing this! Will take a look today and confirm.
